File: Searching and Sorting Algorithms/codigo_cs/13InsertionSort.cs
using System;
using System.Linq;
public class Test13
{
    public static void IS()
    {
        // your code goes here
        int n = int.Parse(Console.ReadLine());
        int[] arr = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);


        for (int i = 1; i < n; i++)
        {
            int curr = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > curr)
            {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = curr;
        }
        Console.WriteLine(string.Join(" ", arr));

    }
}

File: Searching and Sorting Algorithms/13InsertionSort.py
n=int(input())
arr=list(map(int,input().split()))

for i in range(1,n):
    curr=arr[i]
    j=i-1
    while j>=0 and arr[j]>curr:
        arr[j+1]=arr[j]
        j=j-1
    arr[j+1]=curr

print(*arr)